### PR TITLE
[SPARK-40877][DOC][FOLLOW-UP] Update the doc of `DataFrame.stat.crosstab `

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
remove the outdated comments

### Why are the changes needed?
the limitations are not true after [reimplementation](https://github.com/apache/spark/pull/38340)

### Does this PR introduce _any_ user-facing change?
yes

### How was this patch tested?
doc - only

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @HyukjinKwon

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -4217,15 +4217,18 @@ def cov(self, col1: str, col2: str) -> float:
     def crosstab(self, col1: str, col2: str) -> "DataFrame":
         """
         Computes a pair-wise frequency table of the given columns. Also known as a contingency
-        table. The number of distinct values for each column should be less than 1e4. At most 1e6
-        non-zero pair frequencies will be returned.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/dataframe.py:4221]: I think we can just remove this and don't add `.. versionchanged:: 3.4.0`. It's sort of improvement, and trivial to mention.

File: python/pyspark/sql/dataframe.py [hunk 2]
```diff
@@ -4217,15 +4217,18 @@ def cov(self, col1: str, col2: str) -> float:
...
         will be the distinct values of `col2`. The name of the first column will be `$col1_$col2`.
         Pairs that have no occurrences will have zero as their counts.
         :func:`DataFrame.crosstab` and :func:`DataFrameStatFunctions.crosstab` are aliases.
 
         .. versionadded:: 1.4.0
 
+        .. versionchanged:: 3.4.0
+           The underlying implementation was optimized, then there is no limitaions on number of
+           distinct values or pairs.
+
```
Comment (HyukjinKwon) [inline python/pyspark/sql/dataframe.py:4231]:
```suggestion
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM

File: python/pyspark/sql/dataframe.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/dataframe.py:4221]: got it

**Comment by zhengruifeng** [comment]
merged into master

**Pull request closed without merging**